src/http/request_handler.rs

```diff
@@ -132,11 +134,36 @@ fn create_allowed_headers(headers: &HeaderMap, allowed: &BTreeSet<String>) -> He
     new_headers
 }
 
+async fn handle_rest_apis(
+    mut request: Request<Body>,
+    app_ctx: Arc<AppContext>,
+) -> Result<Response<Body>> {
+    *request.uri_mut() = request.uri().path().replace(API_URL_PREFIX, "").parse()?;
+    let req_ctx = Arc::new(create_request_context(&request, app_ctx.as_ref()));
+    if let Some(p_request) = app_ctx.rest_endpoints.matches(&request) {
+        let graphql_request = p_request.to_request(request).await?;
+        let mut response = graphql_request
+            .data(req_ctx.clone())
+            .execute(&app_ctx.schema)
+            .await;
+        response = update_cache_control_header(response, app_ctx.as_ref(), req_ctx);
+        let mut resp = response.to_response()?;
+        update_response_headers(&mut resp, app_ctx.as_ref());
+        return Ok(resp);
+    }
+
+    not_found()
+}
```

Comment:
@coderabbitai there is some duplicate logic in the file. Suggest ways to improve this file structure and reduce duplication.